cabal-install/Distribution/Client/CmdInstall/ClientInstallFlags.hs

```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE LambdaCase #-}
+module Distribution.Client.CmdInstall.ClientInstallFlags
+( BindirMethod(..)
+, ClientInstallFlags(..)
+, defaultClientInstallFlags
+, clientInstallOptions
+) where
+
+import Distribution.Client.Compat.Prelude
+
+import Distribution.ReadE
+         ( ReadE(..), succeedReadE )
+import Distribution.Simple.Command
+         ( ShowOrParseArgs(..), OptionField(..), option, reqArg )
+import Distribution.Simple.Setup
+         ( Flag(..), trueArg, flagToList, toFlag )
+
+import Distribution.Client.InstallSymlink
+         ( OverwritePolicy(..) )
+
+
+data BindirMethod = BindirMethodCopy
+                  | BindirMethodSymlink
+  deriving (Eq, Show, Generic, Bounded, Enum)
+
+instance Binary BindirMethod
+
+data ClientInstallFlags = ClientInstallFlags
+  { cinstInstallLibs :: Flag Bool
+  , cinstEnvironmentPath :: Flag FilePath
+  , cinstOverwritePolicy :: Flag OverwritePolicy
+  , cinstBindirMethod :: Flag BindirMethod --TODO bindir-method or install-method?
```

Comment:
If we're going with `--installdir=...` then this should definitely be `--install-method=` or `--install-policy=` or something else with `--install-*` :-)